assets/src/hooks/useMathWarning.js

```diff
@@ -1,13 +1,20 @@
 import { useEffect, useState } from 'react'
 
-const allAssignmentsHasGoalGrade = assignments =>
-  assignments.filter(a => !(a.goalGradeSetByUser)).length === 0
+// this is the case when either few assignments are graded or few has goal grade.
+// if an assignment is graded then it won't have goal grade and vice versa.
```

Comment:
Would you put the 'vice versa' scenario into the comment line, too?

Reply:
Done